commandline/src/main/java/gov/cms/qpp/conversion/CommandLineRunner.java

```diff
@@ -0,0 +1,194 @@
+
+	private final CommandLine commandLine;
+	private final FileSystem fileSystem;
+	private Set<QrdaScope> scope;
+	private boolean doValidation;
+	private boolean doDefaults;
+	private boolean historical;
+
+	public CommandLineRunner(CommandLine commandLine) {
+		this(commandLine, FileSystems.getDefault());
+	}
+
+	public CommandLineRunner(CommandLine commandLine, FileSystem fileSystem) {
+		Objects.requireNonNull(commandLine, "commandLine");
+		Objects.requireNonNull(fileSystem, "fileSystem");
+
+		this.commandLine = commandLine;
+		this.fileSystem = fileSystem;
+	}
+
+	@Override
+	public void run() {
+		if (isHelp()) {
+			sendHelp();
+			return;
+		}
+
+		if (!hasPotentialFiles()) {
+			DEV_LOG.error("You must specify files to convert");
+			sendHelpHint();
+			return;
+		}
+
+		scope = getScope();
+		if (scope == null) {
+			DEV_LOG.error("A given template scope was invalid");
+			sendHelpHint();
+			return;
+		}
+
+		Set<Path> convert = getConvert();
+
+		List<Path> invalid = convert.stream()
+				.filter(path -> !isValid(path))
+				.collect(Collectors.toList());
+		if (!invalid.isEmpty()) {
+			DEV_LOG.error("Invalid or missing paths: " + invalid);
+			sendHelpHint();
+			return;
+		}
+
+		doValidation = !commandLine.hasOption(CommandLineMain.SKIP_VALIDATION);
+		doDefaults = !commandLine.hasOption(CommandLineMain.SKIP_DEFAULTS);
+		historical = commandLine.hasOption(CommandLineMain.BYGONE);
+
+		convert.parallelStream()
+			.map(ConversionFileWriterWrapper::new)
+			.peek(conversion -> conversion.setContext(createContext()))
+			.forEach(ConversionFileWriterWrapper::transform);
+	}
+
+	private Context createContext() {
+		Context context = new Context();
+		context.setDoDefaults(doDefaults);
+		context.setDoValidation(doValidation);
+		context.setHistorical(historical);
+		context.setScope(scope);
+		return context;
+	}
+
+	private boolean isHelp() {
+		return commandLine.hasOption(CommandLineMain.HELP);
+	}
+
+	private void sendHelp() {
+		StringWriter content = new StringWriter();
+		PrintWriter out = new PrintWriter(content);
+		CommandLineMain.HELP_FORMAT.printHelp(out, CommandLineMain.HELP_FORMAT.getWidth(),
+				"convert", null, CommandLineMain.OPTIONS, 0, 0, null, true);
+
+		DEV_LOG.info(content.toString());
+	}
+
+	private void sendHelpHint() {
+		DEV_LOG.info("Get help by rerunning with -help.");
+	}
+
+	private boolean hasPotentialFiles() {
+		return !commandLine.getArgList().isEmpty();
+	}
+
+	private Set<QrdaScope> getScope() {
+		if (commandLine.hasOption(CommandLineMain.TEMPLATE_SCOPE)) {
+			String[] templateScope = LITERAL_COMMA.split(commandLine.getOptionValue(CommandLineMain.TEMPLATE_SCOPE));
+			Set<QrdaScope> scope = Arrays.stream(templateScope)
+					.map(QrdaScope::getInstanceByName)
+					.filter(Objects::nonNull)
+					.collect(Collectors.toCollection(() -> EnumSet.noneOf(QrdaScope.class)));
+
+			if (scope.size() != templateScope.length) {
+				return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty collection instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1168)